lib/board.rb

```diff
@@ -0,0 +1,74 @@
+class Board
+  def initialize
+    @cells = Array.new(9, " "   )
+  end
+
+  def cells
+    @cells
+  end
+
+  def cells=(array)
+    @cells = array
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def position(pos)
+    # displays position
+    @cells[pos.to_i-1]
+  end
+
+  def full?
+    # returns if board is full
+    @cells.each {|cell|
+      if cell == " "
+        return false
+      end
+    }
+    true
+  end
+
+  def turn_count
+    count = 0
+    @cells.each {|cell|
+      if cell != " "
+        count += 1
+      end
+    }
+    count
```

Comment:
Here, we have another opportunity to use a higher-level, more expressive iterator. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count {|cell| cell != " "}
end
```